src/lib/y2storage/proposal_settings.rb

```diff
@@ -142,204 +237,216 @@ def delete_forced(type)
-      set_from_integer_feature(:btrfs_increase_percentage, :btrfs_increase_percentage)
-
-      set_from_string_feature(:btrfs_default_subvolume, :btrfs_default_subvolume)
-      read_subvolumes_section!
+    # @note When there is no <partitioning> section, legacy format is considered.
+    #
+    # @return [Symbol, nil] :ng or :legacy
+    def features_format
+      return LEGACY_FORMAT if partitioning_section.nil?
+
+      has_ng_subsections = partitioning_section.key?("proposal") && partitioning_section.key?("volumes")
+      has_ng_subsections ? NG_FORMAT : LEGACY_FORMAT
     end
 
-    # New object initialized according to the YaST product features
-    # (i.e. /control.xml)
-    #
-    # @return [ProposalSettings]
-    def self.new_for_current_product
-      settings = new
-      settings.read_product_features!
-      settings
+    def ng_format?
+      format == NG_FORMAT
     end
 
-    def to_s
-      "Storage ProposalSettings\n" \
-        "  use_lvm: #{use_lvm}\n" \
-        "  root_filesystem_type: #{root_filesystem_type}\n" \
-        "  use_snapshots: #{use_snapshots}\n" \
-        "  use_separate_home: #{use_separate_home}\n" \
-        "  home_filesystem_type: #{home_filesystem_type}\n" \
-        "  enlarge_swap_for_suspend: #{enlarge_swap_for_suspend}\n" \
-        "  root_device: #{root_device}\n" \
-        "  candidate_devices: #{candidate_devices}\n" \
-        "  root_base_size: #{root_base_size}\n" \
-        "  root_max_size: #{root_max_size}\n" \
-        "  root_space_percent: #{root_space_percent}\n" \
-        "  btrfs_increase_percentage: #{btrfs_increase_percentage}\n" \
-        "  min_size_to_use_separate_home: #{min_size_to_use_separate_home}\n" \
-        "  btrfs_default_subvolume: #{btrfs_default_subvolume}\n" \
-        "  root_subvolume_read_only: #{root_subvolume_read_only}\n" \
-        "  home_min_size: #{home_min_size}\n" \
-        "  home_max_size: #{home_max_size}\n" \
-        "  subvolumes: \n#{subvolumes}\n"
+    # Sets default values for the settings.
+    # These will be the final values when the setting is not specified in the control file
+    def apply_defaults
+      ng_format? ? apply_ng_defaults : apply_legacy_defaults
     end
 
-    # Check whether using btrfs filesystem with snapshots
+    # Overrides the settings with values read from the YaST product features
+    # (i.e. values in /control.xml).
     #
-    # @return [Boolean]
-    def snapshots_active?
-      root_filesystem_type.is?(:btrfs) && use_snapshots
+    # Settings omitted in the product features are not modified.
+    def load_features
+      ng_format? ? load_ng_features : load_legacy_features
     end
 
-  protected
+    # FIXME: Improve implementation. Use composition to encapsulate logic for
+    # ng and legacy formats
+    def apply_ng_defaults
+      self.lvm                 ||= false
+      self.resize_windows      ||= true
+      # TODO: once other strategies are available, change default value (:use_available)
```

Comment:
This comment ended up in the wrong line. It refers to lvm_vg_strategy, not to windows_delete_mode.

The same happens again in the legacy version of the method